find_email_addresses.py

```diff
@@ -0,0 +1,90 @@
+#
+# Find All Email Adresses - Jana Coding Challenge
+#
+# Chris Anderson
+# 10-11-2016
+#
+
+from bs4 import BeautifulSoup
+import urllib2
+from urlparse import urljoin
+from urlparse import urlparse
+from sets import Set
+import sys
+import re
+
+# Regex from http://scraping.pro/email-validation-regexes/
+CONST_EMAIL_REGEX = "^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"
+DEBUG = False
+
+urls = Set() # visited urls
+emails = Set() # collected emails
+
+# Catch HTTP Error codes silently without crashing
+def open_url(url):
+    try:
+       return urllib2.urlopen(url)
+    except urllib2.HTTPError, err:
+       if err.code == 404:
+            if (DEBUG): print "error 404"
+            return False
+       elif err.code == 403:
+            if (DEBUG): print "error 403"
+            return False
+       else:
+            if (DEBUG): print "other error"
+            return False
+    except urllib2.URLError, err:
+        if (DEBUG): print "other error"
+        return False
+
+# url: current page to parse
+# domain: domain to crawl
+# debug: if true will print each page it visists
+def find_emails(url, domain, debug):
+    html = open_url(url)
+
+    if (html == False): return
```

Comment:
This would probably be cleaner as a multiple-line conditional, rather than two separate lines that handle the condition the same way.